packages/react-dom/src/__tests__/ReactCompositeComponent-test.js

```diff
@@ -1839,12 +1418,24 @@ describe('ReactCompositeComponent', () => {
       );
     }
     const container = document.createElement('div');
+    const root = ReactDOMClient.createRoot(container);
     expect(() => {
-      ReactDOM.render(<Parent />, container);
+      ReactDOMX.flushSync(() => {
+        root.render(<Parent />);
+      });
     }).toErrorDev(
       'Cannot update a component (`A`) while rendering a different component (`B`)',
     );
+
+    // We error, but still update the state.
+    expect(ref.textContent).toBe('1');
+
     // Dedupe.
-    ReactDOM.render(<Parent />, container);
+    ReactDOMX.flushSync(() => {
+      root.render(<Parent />);
+    });
+
+    // We error, but still update the state.
+    expect(ref.textContent).toBe('2');
```

Comment:
This is weird to me. The error says we "cannot update a component while rendering a different component" but the updates do flush? Is that expected @acdlite?